react-components/table-intro/step8.md

```diff
@@ -1,118 +1,64 @@
+In the `src/App.js` file, just below the line that starts with `const App = () => {`, add the following code to the App component. This code creates a state variable called `numPerPage` and sets the default value to `2`, so that we will see 2 rows per page by default.
 
 <pre class="file">
-const [rows, setRows] = React.useState(defaultRows);
+const defaultPerPage = 2;
+const [numPerPage, setNumPerPage] = React.useState(defaultPerPage);
 </pre>
 
-2) Locate the code that sets the rows for the table (`rows={defaultRows}`), and update it to reference the new rows state property you just created.
+<strong>Note:</strong> The second parameter in the destructured array returned by React.useState() is the setter for the state variable you're creating. These "setters" are functions that update the corresponding state value.
+
+2) <strong>Set the `perPage` prop of the `<Pagination />` component to the `numPerPage` state property you just created.</strong>
 
 <pre class="file">
-rows={rows}
+perPage={numPerPage}
 </pre>
 
-<strong>Note: </strong> the itemCount prop will still reference the defaultRows constant as itemCount needs to be aware of the total number of rows in the entire dataset, not the number of rows that's being shown at a given time.
+3) <strong>Create a state property to track the current page of the table data.</strong>
 
-3) Add a function "updateList" just above handlePerPageSelect to handle setting the rows based on when the numPerPage state property changes.
+Add the following code to the App component, just below the previous state property you created. Supply a `1` as the default value so that the table intuitively loads with the first page worth of data.
 
-<pre class="file" data-target="clipboard">
-const updateList = value => {
-  setCurrentPage(value);
-  const beginMark = (value - 1) * numPerPage;
-  const endMark = beginMark + numPerPage;
-  let newRows = defaultRows.slice(beginMark, endMark);
-  setRows(newRows);
-};
+<pre class="file">
+const [currentPage, setCurrentPage] = React.useState(1);
 </pre>
 
-4) Inside the handlePerPageSelect handler, make a call to updateList and pass it the currentPage state property.
+4) <strong>Set the `page` prop of the `<Pagination />` component to the `currentPage` state property.</strong>
 
 <pre class="file">
-updateList(currentPage);
+page={currentPage}
 </pre>
 
-5) Add a useEffect hook inside your component that we'll teach to only run when certain properties change. This effects hook will ultimately ensure `numPerPage` and `rows` stay in sync. Inside this effect, make a call to handlePerPageSelect, pass null for the first param (event) and pass numPerPage as the second param. Place this code just above the return statement in your App component.
+5) <strong>Set the Pagination component dropdown options.</strong>
+
+Set the `perPageOptions` prop of the `<Pagination />` component so that we have two options in the dropdown, 2 and 3. This controls how many rows per page are shown.
```

Comment:
don't use we

Reply:
```suggestion
Set the `perPageOptions` prop of the `<Pagination />` component so that there are two options in the dropdown, 2 and 3. This controls how many rows per page are shown.
```

Reply:
I am not using we. I have corrected this before.